core/metric.go

```diff
@@ -64,6 +67,31 @@ func (n Namespace) Strings() []string {
 	return ns
 }
 
+// getSeparator returns an valid separator from nsPriorityList.
+// Otherwise the core separator is returned.
+func (n Namespace) getSeparator() string {
+	mp := n.toMap()
```

Comment:
This implementation loses order since ranging over a map is not ordered. In the example it's also modifying the global map which means it would only work once.